src/main/java/org/isf/lab/gui/LabEdit.java

```diff
@@ -546,29 +546,23 @@ private JButton getOkButton() {
 				} else if (examSelected.getProcedure() == 3) {
 					lab.setResult(examRowTextField.getText());
 				}
-				boolean result;
 				if (insert) {
 					lab.setAge(tmpAge);
 					try {
-						result = labManager.newLaboratory(lab, labRow);
+						labManager.newLaboratory(lab, labRow);
 					} catch (OHServiceException e1) {
-						result = false;
 						OHServiceExceptionUtil.showMessages(e1);
 					}
 				} else {
 					try {
-						result = labManager.updateLaboratory(lab, labRow);
+						labManager.updateLaboratory(lab, labRow);
+						fireLabUpdated();
+						dispose();
 					} catch (OHServiceException e1) {
-						result = false;
+						MessageDialog.error(null, "angal.common.datacouldnotbesaved.msg");
```

Comment:
shouldn't work this way?
```suggestion
						MessageDialog.error(this, "angal.common.datacouldnotbesaved.msg");
```